packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -97,9 +97,11 @@ export function RepositoryMixin<T extends Class<any>>(superClass: T) {
      * }
      * ```
      */
-    dataSource(dataSource: juggler.DataSource, name?: string) {
+    dataSource(dataSource: Class<juggler.DataSource>, name?: string) {
       const dataSourceKey = `datasources.${name || dataSource.name}`;
```

Comment:
Using `dataSource.name` as the default name value changes the behavior quite a lot.

Before:

```ts
app.dataSource(new DataSource({name: 'db', connector: 'memory'}));
// -> name is "db"
```

After your change:
```
app.bind('datasources.config.db').to(({name: 'db', connector: 'memory'});
app.dataSource(DbDataSource);
// -> name is DbDataSource
```

IMO, the `name` argument must be required when registering a datasource via a class.

Reply:
The `boot` module tries to mitigate this by passing in a `name` to `app.dataSource` where it removes the suffix `DataSource` -- if present.

Reply:
IMO, this API needs to support more than just our `boot` module. We want to encourage people to build their own boot-strapper if they don't like our convention or have a specific use case were our conventions don't work well.

One use case is to write a short script that reproduces certain behavior and calls all LB APIs directly, without going through the boot module. Our API should guide the user to do the right thing, i.e. ask them to provide a datasource name if they are passing a class constructor.